daemon/graphdriver/overlay/copy.go

```diff
@@ -32,7 +33,10 @@ func copyRegular(srcPath, dstPath string, mode os.FileMode) error {
 	}
 	defer dstFile.Close()
 
-	_, err = io.Copy(dstFile, srcFile)
+	bufReader := pools.BufioReader32KPool.Get(srcFile)
```

Comment:
I read this thread https://groups.google.com/forum/#!topic/golang-codereviews/I0FhADsQAb0 and decided that it's really safer to use `bufio.Reader`.